// @holochain-open-dev/membrane-context/dist/membrane-context-provider.js
import { __decorate } from "tslib";
import { css, html, LitElement, property } from 'lit-element';
import { ProviderMixin } from 'lit-element-context';
export class MembraneContextProvider extends ProviderMixin(LitElement) {
    static get provide() {
        return ['membraneContext'];
    }
    updated(changedValues) {
        super.updated(changedValues);
        if (changedValues.has('cellId') ||
            changedValues.has('appWebsocket') ||
            changedValues.has('adminWebsocket')) {
            this.membraneContext = {
                cellId: this.cellId,
                appWebsocket: this.appWebsocket,
                adminWebsocket: this.adminWebsocket,
            };
        }
    }
    static get styles() {
        return css `
      :host {
        display: contents;
      }
    `;
    }
    render() {
        return html `<slot></slot>`;
    }
}
__decorate([
    property({ type: Object })
], MembraneContextProvider.prototype, "membraneContext", void 0);
__decorate([
    property({ type: Array })
], MembraneContextProvider.prototype, "cellId", void 0);
__decorate([
    property({ type: Object })
], MembraneContextProvider.prototype, "appWebsocket", void 0);
__decorate([
    property({ type: Object })
], MembraneContextProvider.prototype, "adminWebsocket", void 0);
